test/testgroups/operatorsourcetests.go

```diff
@@ -0,0 +1,35 @@
+package testgroups
+
+import (
+	"testing"
+
+	"github.com/operator-framework/operator-marketplace/test/helpers"
+	"github.com/operator-framework/operator-marketplace/test/testsuites"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+// OperatorSourceTestGroup creates an OperatorSource and then runs a series of test suites that rely on this resource.
+func OperatorSourceTestGroup(t *testing.T) {
+	// Create a ctx that is used to delete the OperatorSource at the completion of this function.
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get test namespace.
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		t.Errorf("Could not get namespace: %v", err)
+	}
+
+	// Get global framework variables.
+	f := test.Global
+
+	// Create the OperatorSource.
+	err = helpers.CreateRuntimeObject(f, ctx, helpers.CreateOperatorSourceDefinition(namespace))
```

Comment:
So with the change you made, why don't we do this at the beginning of each test suite that needs it?

I'm still thinking about the test that creates an operator source and then deletes it. Creating the operator source here will make that test impossible -- it would be nicer if we could create that test in its own test suite (which means that each test suite that currently exists would need to create an operator source at the start)

Reply:
I don't see how the test you described is limited with the existing approach - the test that creates/deletes an operator source could be done in it's own test suite that is not a part of this group.

Reply:
The original intent with test groups was to prepare resources that could be used across test suites.

Reply:
Oh, this is my bad. I thought this was the harness. Reading comprehension :)